mlflow/pipelines/steps/ingest/datasets.py

```diff
@@ -436,6 +436,14 @@ def __init__(
             self.custom_loader_method_name,
         ) = custom_loader_method.rsplit(".", 1)
 
+    def _validate_user_code_output(self, func, *args):
+        ingested_df = func(*args)
+        if not isinstance(ingested_df, pd.DataFrame):
+            raise MlflowException(
+                message="The `ingested_data` is not a DataFrame, please ingest the correct data format"
```

Comment:
"...please ingest the correct data format" -> "please make sure `self.custom_loader_method_name` returns a Pandas DataFrame object".